newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectSaveChoiceDialog.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+
+import Dialog, { DialogPrimaryButton } from '../../UI/Dialog';
+import { Line } from '../../UI/Grid';
+import FlatButton from '../../UI/FlatButton';
+import LeftLoader from '../../UI/LeftLoader';
+
+type Props = {|
+  onClose: () => void,
+  isLoading: boolean,
+  onSaveAsMainVersion: () => void | Promise<void>,
+  onSaveAsDuplicate: () => void | Promise<void>,
+|};
+
+const CloudProjectRecoveryDialog = ({
+  onClose,
+  isLoading,
+  onSaveAsDuplicate,
+  onSaveAsMainVersion,
+}: Props) => {
+  const actions = [
+    <FlatButton
+      disabled={isLoading}
+      key="save-copy"
+      label={<Trans>Save as copy</Trans>}
+      onClick={onSaveAsDuplicate}
+    />,
+    <LeftLoader isLoading={isLoading} key="save-main">
+      <DialogPrimaryButton
+        primary
+        disabled={isLoading}
+        label={<Trans>Save as main version</Trans>}
+        onClick={onSaveAsMainVersion}
+      />
+    </LeftLoader>,
+  ];
+
+  return (
+    <Dialog
+      open
```

Comment:
same here, adding a cannotBeDismissed={isLoading} to prevent the dialog to be closed mid-save

Reply:
Good idea! In this case, we really want to prevent a closing from the user